setup.py

```diff
@@ -212,9 +215,10 @@ def build_cmake_extension(self, ext: CMakeExtension):
             f"-DCMAKE_MAKE_PROGRAM={ninja_path}",
         ]
         configure_args += (
-            [f"-DPYTHON_EXECUTABLE={sys.executable}"]
-            if platform.system() != "Darwin"
-            else [f"-DPython_EXECUTABLE={sys.executable}"]
+            f"-DPython_ROOT_DIR={sys.prefix}",
+            f"-DPYTHON_VERSION_TO_FIND={sys.version_info.major}.{sys.version_info.minor}",
+            f"-Dnanobind_DIR={nanobind.cmake_dir()}",
+            f"-DNUMPY_INCLUDE_DIR={numpy.get_include()}",
```

Comment:
Just a flyby --- FindPython has support for a numpy target https://cmake.org/cmake/help/latest/module/FindPython.html (e.g. used https://github.com/PennyLaneAI/catalyst/blob/10e33d3f8359056bdc532a7fd4b2617bf30de70c/frontend/catalyst/utils/CMakeLists.txt#L8 and https://github.com/PennyLaneAI/catalyst/blob/10e33d3f8359056bdc532a7fd4b2617bf30de70c/frontend/catalyst/utils/CMakeLists.txt#L57)

Reply:
That's true let me give that a try 👍 

As a side-note, I don't trust CMake's python discovery, it's quite finicky and we've often had problems with it (including this PR). If only we could specify the exact interpreter it should use, and not on a best-effort basis, that would be nice 😅

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1380/commits/3100da44ce748c02111ebcea4736527e246339b0